pkg/api/v1alpha1/cluster.go

```diff
@@ -724,9 +724,13 @@ func validateMirrorConfig(clusterConfig *Cluster) error {
 		return fmt.Errorf("registry mirror port %s is invalid, please provide a valid port", clusterConfig.Spec.RegistryMirrorConfiguration.Port)
 	}
 
-	if !features.RegistryMirrorInsecureSkipVerifySupport().IsActive() {
-		if clusterConfig.Spec.RegistryMirrorConfiguration.InsecureSkipVerify && clusterConfig.Spec.DatacenterRef.Kind != SnowDatacenterKind {
-			return errors.New("insecureSkipVerify is only supported for snow provider")
+	if !features.RegistryMirrorInsecureSkipVerifySupport().IsActive() &&
+		clusterConfig.Spec.RegistryMirrorConfiguration.InsecureSkipVerify {
+		switch clusterConfig.Spec.DatacenterRef.Kind {
+		case NutanixDatacenterKind, SnowDatacenterKind:
+			break
+		default:
+			return errors.New("insecureSkipVerify is only supported for nutanix and snow providers")
```

Comment:
Can we just use the feature flag `REGISTRY_MIRROR_INSECURE_SKIP_VERIFY_SUPPORT` here to bypass the validation for now? I think eventually we could remove it altogether validation once it's successfully supported for all the other providers.

Reply:
Yeah, until it's supported for all providers, this is needed.

Reply:
I think we need to remove the feature flag check as well then. The purpose of it was to have support for snow only, unless that flag was set.